src/Microsoft.AspNet.Mvc.Razor/Resources.resx

```diff
@@ -195,4 +195,9 @@
   <data name="RazorPage_InvalidTagHelperIndexerAssignment" xml:space="preserve">
     <value>Unable to perform '{0}' assignment. Tag helper property '{1}.{2}' must not be null.</value>
   </data>
+  <data name="CouldNotResolveApplicationRelativeUrl_TagHelper" xml:space="preserve">
+    <value>Unexpected return value from '{1}.{2}' for URL '{0}'. If the '{1}' service has been overridden, change '{2}' to replace only the '~/' prefix. Otherwise, add the following directive to the Razor page to disable application-relative URL resolution:
```

Comment:
Discussed the [now-hidden thread](https://github.com/aspnet/Mvc/pull/2875#discussion-diff-35729499) with @Eilon offline. Decided to avoid using "application-relative" in a new way.

Here "application-relative URL resolution" -> "URL resolution relative to the application's 'webroot' setting". Might want to pass "webroot" into the resource.